frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditDate/EditDate.tsx

```diff
@@ -0,0 +1,230 @@
+import { useMemo } from 'react'
+import { Controller, RegisterOptions } from 'react-hook-form'
+import { Box, FormControl, SimpleGrid } from '@chakra-ui/react'
+import { isBefore, isDate, isEqual } from 'date-fns'
+import { extend, get, isEmpty, pick } from 'lodash'
+
+import {
+  DateFieldBase,
+  DateSelectedValidation,
+  DateValidationOptions,
+} from '~shared/types/field'
+
+import {
+  transformDateToShortIsoString,
+  transformShortIsoStringToDate,
+} from '~utils/date'
+import { createBaseValidationRules } from '~utils/fieldValidation'
+import DateInput from '~components/DatePicker'
+import { SingleSelect } from '~components/Dropdown'
+import FormErrorMessage from '~components/FormControl/FormErrorMessage'
+import FormLabel from '~components/FormControl/FormLabel'
+import Input from '~components/Input'
+import Textarea from '~components/Textarea'
+import Toggle from '~components/Toggle'
+
+import { DrawerContentContainer } from '../common/DrawerContentContainer'
+import { FormFieldDrawerActions } from '../common/FormFieldDrawerActions'
+import { EditFieldProps } from '../common/types'
+import { useEditFieldForm } from '../common/useEditFieldForm'
+
+type EditDateProps = EditFieldProps<DateFieldBase>
+
+const EDIT_DATE_FIELD_KEYS = ['title', 'description', 'required'] as const
+
+type EditDateInputs = Pick<
+  DateFieldBase,
+  typeof EDIT_DATE_FIELD_KEYS[number]
+> & {
+  dateValidation: {
+    selectedDateValidation: DateSelectedValidation | ''
+    customMaxDate: string
+    customMinDate: string
+  }
+}
+
+const transformDateFieldToEditForm = (field: DateFieldBase): EditDateInputs => {
+  const nextValidationOptions = {
+    selectedDateValidation:
+      field.dateValidation.selectedDateValidation ?? ('' as const),
+    customMaxDate:
+      (!!field.dateValidation.selectedDateValidation &&
+        transformDateToShortIsoString(field.dateValidation.customMaxDate)) ||
+      ('' as const),
```

Comment:
Considering what this does, a ternary might be clearer, and same for `customMinDate` below
```suggestion
    customMaxDate: (field.dateValidation.selectedDateValidation
        ? transformDateToShortIsoString(field.dateValidation.customMaxDate)
        : ('' as const)),
```

Reply:
transformDateToShortIsoString can return `null`. added an extra nullish check, think your suggestion with a nullish check is still clearer